### PR TITLE
The hat barrel

### PR DESCRIPTION
**Pull request by Fueredoriku**
- Added model, Texture and animation
- Made prefab that can be succesfully instantiated by gunfactory
- Modifier system fixed in some aspects (see commits for more details)
- Hat barrel implemented as an item and added to relevant auctionStage
- Hat bullet sticks to players they hit

Misc:
- Updated guntest scene with a dummy with hitbox and healthcontroller to test new augments more efficently

hatSlingerAnimFin

The multiple onHit functionality of the hats attached to victim will be implemented in it's own separate PR.

### CODE REVIEW

**Comment by toberge** [comment]
The commit history seems messed up?

File: Assets/Scripts/Augment/ProjectileModifier.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 /// <summary>
 /// A class that can be extended in order to give a behavioural property that can be applied to any projectile created by any set of parts.
-/// Why? Because putting a copy of every needed bullet script on every single bullet that is created is ver unefficient.
+/// Why? Because putting a copy of every needed bullet script on every single bullet that is created is very unefficient.
```
Comment (toberge) [inline Assets/Scripts/Augment/ProjectileModifier.cs:7]: inefficient ;)

**Comment by toberge** [comment]
Glanced at your code and it is good to see some better structure for augment scripts. I have some feedback regarding naming:

- Why BarrelHat and not HatBarrel?
- In the same vein, why `bullet_hat` and `bulletModel_hat`? Why are we using `snake_case_withCamels` in the first place? (I know I added sth with `snake_case` myself, I'm still not on board with it)
- RotatorModifier should be RotationModifier since it modifies rotation, not a rotator.

File: Assets/Scripts/Augment/BulletModifiers/StickyProjectileModifier.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    // Does stuck object trigger onHit?
+    [SerializeField]
+    private bool triggerOnHit = true;
+    // How often does it trigger onHit (in seconds)?
+    [SerializeField]
+    private float onHitInterval = 1f;
+
+    public void StickToTarget(HitboxController controller, ref ProjectileState state, GunStats stats)
+    {
+        var stuck = Instantiate(stuckObject, state.position, state.rotation, controller.transform);
```
Comment (toberge) [inline Assets/Scripts/Augment/BulletModifiers/StickyProjectileModifier.cs:24]: Perhaps adjust the position before instancing? Sometimes you only see the brim of the hat:

image

**Review by toberge — CHANGES_REQUESTED** [review]
`bullet` folder in `Prefabs/GunParts` should be `Bullets`

Otherwise this looks quite good, my dude

File: Assets/Scripts/Augment/BulletModifiers/StickyProjectileModifier.cs (hunk 2 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/BulletModifiers/StickyProjectileModifier.cs:24]: Hope this satisfies all hat-pincushion-related needs 🤠
image

**Review by toberge — APPROVED** [review]
I see my requests have been fulfilled. We should have a discussion about asset naming convention... _outside_ this PR.

**Comment by Fueredoriku** [comment]
> We should have a discussion about asset naming convention... _outside_ this PR.

Yup, we'll try to cover #19 at todays meeting

**Pull request merged**